src/cartesian/Bar.js

```diff
@@ -262,10 +263,16 @@ class Bar extends Component {
               }
 
               if (layout === 'horizontal') {
-                const interpolator = interpolateNumber(0, entry.height);
-                const h = interpolator(t);
+                const interpolatorX = interpolateNumber(width * 2, entry.x);
```

Comment:
@huei-tan-sociomantic Maybe you can add some comments here.

Reply:
DONE.